src/AbstractOperations/binary_operations.jl

```diff
@@ -187,6 +187,36 @@ macro binary(ops...)
     return expr
 end
 
+#####
+##### BinaryOperations with ZeroField
+#####
+
+import Base: +, -, *, /, ==
+using Oceananigans.Fields: ZeroField
+
+==(::ZeroField, ::ZeroField) = true
+
++(a::ZeroField, b::AbstractField) = b
++(a::AbstractField, b::ZeroField) = +(b, a)
++(a::ZeroField, b::Number) = b
++(a::Number, b::ZeroField) = +(b, a)
+
+-(a::ZeroField, b::AbstractField) = -b
+-(b::AbstractField, a::ZeroField) = b
+-(a::ZeroField, b::Number) = -b
+-(b::Number, a::ZeroField) = b
+
+*(a::ZeroField, b::AbstractField) = a
+*(b::AbstractField, a::ZeroField) = *(a, b)
+*(a::ZeroField, b::Number) = a
+*(b::Number, a::ZeroField) = *(a, b)
+
+/(a::ZeroField, b::AbstractField) = a
+/(b::AbstractField, a::ZeroField) = Inf
+/(a::ZeroField, b::Number) = a
+/(b::Number, a::ZeroField) = Inf
```

Comment:
Shouldn't this be `ConstantField(Inf)`?

Reply:
Also, I just realized that it'd probably be good if we made stuff like the operation below work too, no?

```julia
julia> 1 + ConstantField(1)
ERROR: type ConstantField has no field grid
```

Reply:
Sorry I missed this. I think it's better not to use the `ConstantField` wrapper when it can be avoided, since this will yield simpler expression trees?